src/payment/strategies/quadpay/quadpay-payment-strategy.ts

```diff
@@ -35,13 +35,13 @@ export default class QuadpayPaymentStrategy implements PaymentStrategy {
             throw new PaymentArgumentInvalidError(['payment']);
         }
 
-        let nonce: string;
+        let nonce: { id: string; uri: string };
         const { methodId } = payment;
         const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId, options));
         const { clientToken = '' } = state.paymentMethods.getPaymentMethodOrThrow(methodId);
 
         try {
-            ({ id: nonce } = JSON.parse(clientToken));
+            nonce = JSON.parse(clientToken);
```

Comment:
Please see https://github.com/bigcommerce/checkout-sdk-js/pull/1185/files#r672792219.

Reply:
Sure, We are using this approach due to that the zip/quadpay dev team has removed the `uri` data from the `Retrieve checkout` endpoint(the `get` call to the ` /checkouts/{Id}`), and the only way to get this redirect url is from the initialization call `POST` to `/checkouts`, so, we are passing this `uri` data through the Payment initialization call(where we are getting the `client_token`) to get the `uri` data and then redirect to the external zip/quadpay checkout.

Reply:
Thanks for the explanation. Please see https://github.com/bigcommerce/checkout-sdk-js/pull/1185/files#r673701735

Reply:
I saw you made changes for the other PR regarding the uri implementation, can that be done here as well?

Reply:
Hello @animesh1987 I'm going to update this Pull request and notify you ASAP, I'm creating the `bigcommerce php` PR with the changes to support `getInitializationData` in the quadpay module.